ui/app/src/components/DateTimePicker.tsx

```diff
@@ -100,97 +120,117 @@ interface timezoneType {
 
 const timezones = getTimezones();
 
-const DateTimePicker = withStyles(dateTimePickerStyles)((props: DateTimePickerProps) => {
-  const {
-    classes,
-    onChange,
-    onChangeDate,
-    onChangeTime,
-    onChangeTimezone,
-    initialDate = moment(),
-    timezone = moment.tz.guess(),
-    dateFormat = 'YYYY-MM-DD',
-    timeFormat = 'HH:MM:ss',
-    controls = ['date', 'time', 'timezone']
-  } = props;
-
-  let initialDateMoment;
+const getDateMoment = (dateString, timezoneObj) => {
+  let dateMoment;
 
-  switch(typeof initialDate) {
+  switch (typeof dateString) {
     case 'string':
     case 'number':
-      initialDateMoment = moment(initialDate);
+      dateMoment = timezoneObj ? moment.tz(dateString, timezoneObj.timezoneName) : moment.tz(dateString);
       break;
     case 'object':
       // moment object, stays the same
-      initialDateMoment = initialDate;
+      dateMoment = dateString;
       break;
     default:
-      initialDateMoment = moment();
+      dateMoment = moment();
+  }
+
+  return dateMoment;
+};
+
+const DateTimePicker = withStyles(dateTimePickerStyles)((props: DateTimePickerProps) => {
+  const {
+    id,
+    classes,
+    onChange,
+    date = moment(),
+    disabled = false,
+    controls = ['date', 'time', 'timezone'],
+    datePickerProps = {},
+    timePickerProps = {},
+    timeZonePickerProps = {}
+  } = props;
+
+  // Set defaults on wrapped controls props
+  if (nou(timeZonePickerProps.timezone)) {
+    timeZonePickerProps['timezone'] = moment.tz.guess();
+  }
+  if (nou(datePickerProps.dateFormat)) {
+    datePickerProps['dateFormat'] = 'YYYY-MM-DD';
+  }
+  if (nou(datePickerProps.disablePast)) {
+    datePickerProps['disablePast'] = false;
+  }
+  if (nou(timePickerProps.timeFormat)) {
+    timePickerProps['timeFormat'] = 'HH:MM:ss';
   }
 
-  const timezoneObj = timezones.find( tz => (tz.timezoneName === timezone) );
-  const [ selectedDateTime, setSelectedDateTime ] = useState(initialDateMoment);
-  const [ selectedTimezone, setSelectedTimezone ] = useState(timezoneObj);
+  let dateMoment;
+  let timezoneObj = timezones.find(tz => (tz.timezoneName === unescape(timeZonePickerProps.timezone)));
+  dateMoment = getDateMoment(date, timezoneObj);
+  const { formatMessage } = useIntl();
 
-  const handleDateChange = (name: string) => (date: Date | null) => {
-    let updatedDateTime = selectedDateTime;
+  const handleDateChange = (name: string) => (newDate: Date | null) => {
+    let updatedDateTime = dateMoment;
```

Comment:
Why use the closured `dateMoment`? 

If you're hoping the component reflects this by updating the same var, it's not the right data flow. This also leaves time out of the update. You already have a date object (`newDate`). Just create a new moment with it.